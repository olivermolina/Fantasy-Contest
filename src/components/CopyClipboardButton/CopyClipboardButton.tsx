import React, { useState } from 'react';
import { Tooltip } from '@mui/material';
import Icons from '~/components/Icons/Icons';

interface CopyButtonProps {
  text: string;
}

const CopyClipboardButton = (props: CopyButtonProps) => {
  const [open, setOpen] = useState(false);
  const handleClick = () => {
    navigator.clipboard.writeText(props.text);
    setOpen(true);
  };
  const handleClose = () => {
    setOpen(false);
  };
  return (
    <Tooltip title="Copied to clipboard!" open={open} onClose={handleClose}>
      <button onClick={handleClick}>
        <Icons.ClipboardDocument className={'h-6 w-6'} />
      </button>
    </Tooltip>
  );
};

export default CopyClipboardButton;
