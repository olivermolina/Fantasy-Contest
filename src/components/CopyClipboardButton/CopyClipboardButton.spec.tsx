import React from 'react';
import { render } from '@testing-library/react';
import CopyClipboardButton from './CopyClipboardButton';

describe('CopyClipboardButton', () => {
  it('should render', () => {
    const { container } = render(
      <CopyClipboardButton text={'Text copied to clipboard!'} />,
    );
    expect(container).toMatchSnapshot();
  });
});
