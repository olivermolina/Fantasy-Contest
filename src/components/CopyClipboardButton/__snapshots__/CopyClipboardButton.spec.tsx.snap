// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyClipboardButton should render 1`] = `
<div>
  <button
    aria-label="Copied to clipboard!"
    class=""
    data-mui-internal-clone-element="true"
  >
    <svg
      class="h-6 w-6"
      data-file-name="SvgClipboardDocument"
    />
  </button>
</div>
`;
